File: content/search/source/includes/query/operators-collectors/autocomplete/code-snippets/kotlin/tutorial-multi.kt
import com.mongodb.client.model.Aggregates.limit
import com.mongodb.client.model.Aggregates.project
import com.mongodb.client.model.Projections.*
import com.mongodb.kotlin.client.coroutine.MongoClient
import kotlinx.coroutines.runBlocking
import org.bson.Document

fun main() {
    // connect to your Atlas cluster
    val uri = "<connection-string>"
    val mongoClient = MongoClient.create(uri)

    // set namespace
    val database = mongoClient.getDatabase("sample_mflix")
    val collection = database.getCollection<Document>("movies")

    runBlocking {
        // define pipeline
        val agg = Document(
            "\$search",
            Document("index", "autocomplete-tutorial")
                .append(
                    "compound",
                    Document(
                        "should", listOf(
                            Document(
                                "autocomplete",
                                Document("query", "pri")
                                    .append("path", "title")
                            ),
                            Document(
                                "autocomplete",
                                Document("query", "pri")
                                    .append("path", "plot")
                            )
                        )
                    )
                        .append("minimumShouldMatch", 1L)
                )
        )

        // run pipeline and print results
        val resultsFlow = collection.aggregate<Document>(
            listOf(
                agg,
                limit(5),
                project(fields(excludeId(), include("title", "plot")))
            )
        )
        resultsFlow.collect { println(it) }
    }
    mongoClient.close()
}

File: content/search/source/includes/query/operators-collectors/autocomplete/code-snippets/nodejs/tutorial-multi.js
const { MongoClient } = require("mongodb");

// connect to your Atlas cluster
const uri =
    "<connection-string>";

const client = new MongoClient(uri);

async function run() {
    try {
        await client.connect();

        // set namespace
        const database = client.db("sample_mflix");
        const coll = database.collection("movies");

        // define pipeline
        const agg = [
            {
                '$search': {
                    'index': 'autocomplete-tutorial',
                    'compound': {
                        'should': [
                            {
                                'autocomplete': {
                                    'query': 'pri', 
                                    'path': 'title'
                                }
                            }, 
                            {
                                'autocomplete': {
                                    'query': 'pri', 
                                    'path': 'plot'
                                }
                            }
                        ], 
                        'minimumShouldMatch': 1
                    }
                }
            }, 
            {
                '$limit': 5
            }, 
            {
                '$project': {
                    '_id': 0, 
                    'title': 1, 
                    'plot': 1
                }
            }
        ];
        // run pipelines
        const result = await coll.aggregate(agg);

        // print results
        await result.forEach((doc) => console.log(doc));
    } finally {
        await client.close();
    }
}
run().catch(console.dir);
